iib/workers/tasks/build.py

```diff
@@ -755,25 +755,55 @@ def _adjust_operator_manifests(manifests_path):
     For any container image pull spec found in the Operator CSV files, replace floating
     tags with pinned digests, e.g. `image:latest` becomes `image@sha256:...`.
 
+    If spec.relatedImages is not set, it will be set with the pinned digests. If it is set but
+    there are also RELATED_IMAGE_* environment variables set, an exception will be raised.
+
     This method relies on the OperatorManifest class to properly identify and apply the
     modifications as needed.
 
     :param str manifests_path: the full path to the directory containing the operator manifests.
+    :raises IIBError: if the operator manifest has invalid entries
     """
     operator_manifest = OperatorManifest.from_directory(manifests_path)
     found_pullspecs = set()
+    operator_csvs = []
     for operator_csv in operator_manifest.files:
+        if operator_csv.has_related_images():
+            csv_file_name = os.path.basename(operator_csv.path)
+            if operator_csv.has_related_image_envs():
+                raise IIBError(
+                    f'The ClusterServiceVersion file {csv_file_name} has entries in '
+                    'spec.relatedImages and one or more containers have RELATED_IMAGE_* '
+                    'environment variables set. This is not allowed for bundles regenerated with '
+                    'IIB.'
+                )
+            log.debug(
+                'Skipping pinning since the ClusterServiceVersion file %s has entries in '
```

Comment:
mprahl++ for using the entire name of the CSV file everywhere. It is very confusing with just the acronym! :D

Reply:
Should this log statement state that you are skipping pinning now and will do it later for these CSV's if they are not pinned to their digest? Am I missing something?

Reply:
Based on what @lcarva said, if `spec.relatedImages` is set, then don't do any changes to the CSV file.